File: socks/src/main/java/com/example/socks/utils/Operation.java
package com.example.socks.utils;

public enum Operation {
    MORE_THAN("moreThan"),
    LESS_THAN("lessThan"),
    EQUAL("equal");

    private String operation;

    Operation(String operation)
    {
        this.operation = operation;
    }

    public String getOperation() {
        return operation;
    }
}

File: src/main/kotlin/com/vitekkor/socksShop/data/Operation.kt
package com.vitekkor.socksShop.data

enum class Operation {
    MORE_THAN,
    LESS_THAN,
    EQUAL;

    companion object {
        fun getFromStringOrNull(string: String): Operation? {
            return when (string) {
                "moreThan" -> MORE_THAN
                "lessThan" -> LESS_THAN
                "equal" -> EQUAL
                else -> null
            }
        }
    }
}
